### PR TITLE
ChannelFinder

### LINKED ISSUE #17: Package ChannelFinder
This includes:

- the ChannelFinder server: https://github.com/ChannelFinder/ChannelFinderService
- the RecSync RecCeiver client: https://github.com/ChannelFinder/recsync/tree/master/client
- the RecSync RecCaster server: https://github.com/ChannelFinder/recsync/tree/master/server
- maybe the cfNameserver: https://github.com/ChannelFinder/cfNameserver

### PR DESCRIPTION
**Pull request by minijackson**
Fixes #17 

- [x] ChannelFinder service package
- [x] ChannelFinder Python module package (needed by RecCeiver)
- [x] RecCeiver package
- [x] RecCaster support module package
- [x] ChannelFinder NixOS module
- [x] RecCeiver NixOS module
- [x] NixOS integration test
- [x] User-guide for integrating RecCaster in an IOC
- [x] User-guide for setting up a ChannelFinder service + RecCeiver server

I did not document the "embedded LDAP" of the ChannelFinder service, since I've had some issues with it.

### CODE REVIEW

File: docs/ioc/user-guides/reccaster.rst [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+set it as environment variable
+and expose it using the ``addReccasterEnvVars`` function.
+For example:
+
+.. code-block:: csh
+   :caption: :file:`iocBoot/ioc{Example}/st.cmd`
+
+   epicsEnvSet("CONTACT", "mycontact")
+   addReccasterEnvVars("CONTACT")
+
```
Comment (agaget) [inline docs/ioc/user-guides/reccaster.rst:95]: some exist by default and are used by Phoebus: 
epicsEnvSet("IOCNAME", "abc-ioc1")
epicsEnvSet("ENGINEER", "Fred")
epicsEnvSet("LOCATION", "Rack 123")

It is mentionned in the example after but should be mentionned here also

File: docs/nixos-services/user-guides/channel-finder.rst [hunk 2]
```diff
@@ -0,0 +1,327 @@
...
+           };
+         };
+       };
+     };
+   };
+
+External ChannelFinder server
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+If you ChannelFinder server is located on another machine,
```
Comment (agaget) [inline docs/nixos-services/user-guides/channel-finder.rst:306]: If youR ChannelFinder with an R

File: docs/ioc/user-guides/reccaster.rst (hunk 1 above)
Comment (minijackson) [reply docs/ioc/user-guides/reccaster.rst:95]: Used by Phoebus, as in their are displayed specially?

Comment (agaget) [reply docs/ioc/user-guides/reccaster.rst:95]: yes
image-20230616152439-1

see here the columns ENGINEER, IOCNAME. Location is probably hidden in the right

Comment (minijackson) [reply docs/ioc/user-guides/reccaster.rst:96]: I can't reproduce with those variables. With an empty `environment_vars` in the RecCeiver config, I get:

image

I think I'll document `hostName`, `iocName`, and `iocid` then

Comment (agaget) [reply docs/ioc/user-guides/reccaster.rst:96]: i don't understand what I am looking at

Comment (minijackson) [reply docs/ioc/user-guides/reccaster.rst:96]: Ah, I see I think the issue, I'll update the doc.

Comment (minijackson) [reply docs/ioc/user-guides/reccaster.rst:96]: Done

**Review by agaget — APPROVED** [review]

**Review by stephane-cea — APPROVED** [review]
Looks good to me!

**Pull request merged**